plugins/store/src/store.rs

```diff
@@ -12,39 +12,39 @@ use std::{
     sync::{Arc, Mutex},
     time::Duration,
 };
-use tauri::{AppHandle, Emitter, Manager, Resource, Runtime};
+use tauri::{path::BaseDirectory, AppHandle, Emitter, Manager, Resource, ResourceId, Runtime};
 use tokio::{
     select,
     sync::mpsc::{unbounded_channel, UnboundedSender},
     time::sleep,
 };
 
-type SerializeFn =
+pub type SerializeFn =
     fn(&HashMap<String, JsonValue>) -> Result<Vec<u8>, Box<dyn std::error::Error + Send + Sync>>;
-pub(crate) type DeserializeFn =
+pub type DeserializeFn =
     fn(&[u8]) -> Result<HashMap<String, JsonValue>, Box<dyn std::error::Error + Send + Sync>>;
 
-fn default_serialize(
-    cache: &HashMap<String, JsonValue>,
-) -> Result<Vec<u8>, Box<dyn std::error::Error + Send + Sync>> {
-    Ok(serde_json::to_vec(&cache)?)
-}
-
-fn default_deserialize(
-    bytes: &[u8],
-) -> Result<HashMap<String, JsonValue>, Box<dyn std::error::Error + Send + Sync>> {
-    serde_json::from_slice(bytes).map_err(Into::into)
+pub(crate) fn resolve_store_path<R: Runtime>(
+    app: &AppHandle<R>,
+    path: impl AsRef<Path>,
+) -> PathBuf {
+    dunce::simplified(
+        &app.path()
+            .resolve(path, BaseDirectory::AppData)
+            .expect("failed to resolve app dir"),
```

Comment:
no expect, return a result instead

Reply:
That means we need to return result in `StoreBuilder::new` and all the related places, will this ever fail?

Reply:
we can delay resolving the path until `StoreBuilder::build` and there we can return a result.

Reply:
I don't know if we want to make `app.store("some-store")` return a result to be honest, also what about `get_store`?

Reply:
any function that calls `resolve_store_path` should propagate the error, we can't ensure `app.path().resolve()` will always work.